### DISCUSSION TITLE
Maximise self consumption broken

### DISCUSSION THREAD

**Discussion by stefaanv** [wlcrs/huawei_solar#342] [General]
After installing this integration and fiddling around with the settings, my battery is no longer discharging when there is little solar power, as used to be the case before, and which is needed to maximise self consumption.  The solar power is charged into the battery and the house consumption is taken directly from the grid instead of the battery
 
![image](https://user-images.githubusercontent.com/18347072/227794957-fcd60e17-5542-4223-8828-8eaf4aa80236.png)
Any ideas about how to set the parameters to restore maximum self consumptions ?  Below the current setup.  
![image](https://user-images.githubusercontent.com/18347072/227794811-c453d857-54f4-4059-87cc-245e5d1783a0.png)

**Comment by g3r0nim00** [comment]
Can you add some more details about your setup? Inverter version & software release, battery & software release, ...?
I have a 6KTL-M1 & 10kWh LUNA setup which is running fine in maximise self consumption mode...
There used to be a bug in the Huawei software which caused the battery to stop discharging, but that has been fixed some (Huawei) releases ago.
Can you also confirm the settings in the Fusion solar portal/app are 'as expected'?

**Comment by stefaanv** [comment]
It looks that by some kind of extreme coincidence, one of the batteries dicided to calibrate itself at exactly the same time that I started using this HASS integration.
I'll follow this up one more day to see if everything works like before.
